File: Tema 5 - Arrrays/ejercicio8/archives/ingresarDato/ingresarDato.py
"""
Este módulo muestra un MENSAJE incrementando el VALOR de i.
"""
import sys
import json

if len(sys.argv) > 1:
    NUMERO = json.loads(sys.argv[1])
    MENSAJE = ""
    ARRAYVALIDO = True
    for elemento in NUMERO:
        if elemento < 0:
            ARRAYVALIDO = False
            break
    if ARRAYVALIDO:
        COLORCAMBIO1 = "background-COLOR: #2EFE64; COLOR:black"
        COLORCAMBIO2 = "background-COLOR: #FE2E2E; COLOR:black"
        auxPri = []
        auxNoPri = []
        aux = []
        auxOrde = []

        def es_primo(numero2):
            """
            Determina si un número es primo.

            Args:
                numero2 (int): El número que se va a verificar.

            Returns:
                bool: True si el número es primo, False en caso contrario.
            """
            if numero2 < 2:
                return False
            elif numero2 == 2 or numero2 == 3:
                return True
            elif numero2 % 2 == 0:
                return False
            else:
                for i in range(2, int(numero2**0.5) + 1):
                    if numero2 % i == 0:
                        return False
                return True

        for elemento in NUMERO:
            if es_primo(elemento):
                auxPri.append(elemento)
            else:
                auxNoPri.append(elemento)
        aux = auxPri + auxNoPri
        auxPri.sort()
        auxNoPri.sort()
        auxOrde = auxPri + auxNoPri
        MENSAJE = "<table border='1' >"
        filas = [
            ["Indice", list(range(len(NUMERO)))],
            ["Matriz", NUMERO],
            ["Aux", aux],
            ["Ordenado", auxOrde],
        ]
        for fila in filas:
            TITULO = fila[0]
            datos = fila[1]
            MENSAJE += "<tr>"
            MENSAJE += f"<th>{TITULO}</th>"
            if TITULO != "Indice":
                for elemento in datos:
                    COLOR = COLORCAMBIO1 if es_primo(elemento) else COLORCAMBIO2
                    MENSAJE += f"<td style='{COLOR}'>{elemento}</td>"
            else:
                for elemento in datos:
                    MENSAJE += f"<td>{elemento}</td>"
            MENSAJE += "</tr>"
        MENSAJE += "</table>"
        MENSAJE += "El array tiene:<br>"
        PLURALPRI = (
            " numeros primos y <br>" if len(auxPri) != 1 else " numero primo y <br>"
        )
        PLURALNOPRI = (
            " numeros no primos <br>" if len(auxNoPri) != 1 else " numero no primo <br>"
        )
        MENSAJE += f"{len(auxPri)}{PLURALPRI}"
        MENSAJE += f"{len(auxNoPri)}{PLURALNOPRI} "
    print(f"{MENSAJE}")

File: Tema 5 - Arrrays/ejercicio8/archives/ingresarDato/ingresarDato.php
<?php
if (isset($_GET["numeros"])) {
    $numeros = json_decode($_GET["numeros"]);
    if (count($numeros) === 5) {
        $arrayValido = true;
        foreach ($numeros as $elemento) {
            if ($elemento < 0) {
                $arrayValido = false;
                break;
            }
        }
        if ($arrayValido) {
            $mensaje = "";
            $colorCambio1 = 'background-color: #2EFE64; color:black';
            $colorCambio2 = 'background-color: #FE2E2E; color:black';
            $auxPri = [];
            $auxNoPri = [];
            $aux = [];
            $auxOrde = [];
            function esPrimo($numero)
            {
                if ($numero < 2) {
                    return false;
                } elseif ($numero == 2 || $numero == 3) {
                    return true;
                } elseif ($numero % 2 == 0) {
                    return false;
                } else {
                    for ($i = 2; $i * $i <= $numero; $i++) {
                        if ($numero % $i == 0) {
                            return false;
                        }
                    }
                    return true;
                }
            }

            foreach ($numeros as $elemento) {
                if (esPrimo($elemento)) {
                    array_push($auxPri, $elemento);
                } else {
                    array_push($auxNoPri, $elemento);
                }
            }
            $aux = array_merge($auxPri, $auxNoPri);
            sort($auxPri);
            sort($auxNoPri);
            $auxOrde = array_merge($auxPri, $auxNoPri);
            $mensaje = "<table border='1' >";
            $filas = [
                ["Índice", array_keys($numeros)],
                ["Matriz", $numeros],
                ["Aux", $aux],
                ["Ordenado", $auxOrde]
            ];

            foreach ($filas as $fila) {
                $titulo = $fila[0];
                $datos = $fila[1];

                $mensaje .= "<tr>";
                $mensaje .= "<th>$titulo</th>";

                if ($titulo !== "Índice") {
                    foreach ($datos as $elemento) {
                        $color = esPrimo($elemento) ? $colorCambio1 : $colorCambio2;
                        $mensaje .= "<td style='$color'>$elemento</td>";
                    }
                } else {
                    foreach ($datos as $indice) {
                        $mensaje .= "<td>$indice</td>";
                    }
                }

                $mensaje .= "</tr>";
            }

            $mensaje .= "</table>";

            $mensaje .= "El array tiene:<br>";
            $mensaje .= count($auxPri) . ((count($auxPri) != 1) ? " numeros primos y <br>" : " numero primo y <br>");
            $mensaje .= count($auxNoPri) . ((count($auxNoPri) != 1) ? " numeros no primos" : " numero no primo");
            echo $mensaje;
        } else {
            echo "<br><br>El array tiene numeros  negativos";
        }
    }
}
